### PR TITLE
Restructure MacOS framework package to fix malformed Framework errors

### PR DESCRIPTION
**Pull request by vraspar**
### Description

Refactor framework directory structure for MacOS packages

### Motivation and Context

Apple started enforcing specific [framework structure](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/FrameworkAnatomy.html) for MacOS packages. We need to change how we package for MacOS to follow the guidelines

Fixes following issue: [Malformed Framework](https://github.com/microsoft/onnxruntime-swift-package-manager/issues/19
)

### CODE REVIEW

**Comment by vraspar** [comment]
[Artifacts](https://aiinfra.visualstudio.com/Lotus/_build/results?buildId=519635&view=artifacts&pathAsName=false&type=publishedArtifacts)

**Comment by skottmckay** [comment]
Based on the zip file size from a build of `main` vs the branch it looks like 'Current' is being stored as a copy in the zip instead of a symlink. Is there an option we need to specify to avoid that? 

image

**Comment by edgchen1** [comment]
> Based on the zip file size from a build of `main` vs the branch it looks like 'Current' is being stored as a copy in the zip instead of a symlink. Is there an option we need to specify to avoid that?
> 
> image

yeah, looks like this `zip` call should be updated to allow symlinks:

https://github.com/microsoft/onnxruntime/blob/d8888136e3cdf29fa63d3b0a08a58683a7c9f0a0/tools/ci_build/github/apple/assemble_apple_packaging_artifacts.sh#L29

File: tools/ci_build/github/apple/build_apple_framework.py [hunk 1]
```diff
@@ -181,6 +205,26 @@ def _build_package(args):
 
     subprocess.run(build_xcframework_cmd, shell=False, check=True, cwd=REPO_DIR)
 
+    # for macos-arm64_x86_64 create symbloc links
```
Comment (skottmckay) [inline tools/ci_build/github/apple/build_apple_framework.py:208]:
```suggestion
    # for macos-arm64_x86_64 create symbolic links
```

File: tools/ci_build/github/apple/build_apple_framework.py [hunk 2]
```diff
@@ -89,18 +89,41 @@ def _build_for_apple_sysroot(
...
-
-    # use lipo to create a fat ort library
-    lipo_command = ["lipo", "-create"]
-    lipo_command += ort_libs
-    lipo_command += ["-output", os.path.join(framework_dir, "onnxruntime")]
-    subprocess.run(lipo_command, shell=False, check=True)
+
+    # macos requires different framework structure:
+    # https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPFrameworks/Concepts/FrameworkAnatomy.html
+    if sysroot == "macosx":
```
Comment (skottmckay) [inline tools/ci_build/github/apple/build_apple_framework.py:95]: Should this also include "macabi" for mac-catalyst builds?

**Review by skottmckay — APPROVED** [review]

File: tools/ci_build/github/apple/assemble_apple_packaging_artifacts.sh [hunk 3]
```diff
@@ -23,10 +23,36 @@ ORT_POD_VERSION=${4:?${USAGE_TEXT}}
 POD_ARCHIVE_BASENAME="pod-archive-${POD_NAME}-${ORT_POD_VERSION}.zip"
 PODSPEC_BASENAME="${POD_NAME}.podspec"
 
+
+# Macos requires a different structure for the framework
+# This will create the necessary symlinks for the macos framework before packaging
+# Adding the symlinks here rather than in the build script ensures that symlinks are not lost
```
Comment (edgchen1) [inline tools/ci_build/github/apple/assemble_apple_packaging_artifacts.sh:29]: is it `xcodebuild -create-xcframework` which loses the symlinks?

**Review by edgchen1 — APPROVED** [review]

File: tools/ci_build/github/apple/assemble_apple_packaging_artifacts.sh (hunk 3 above)
Comment (vraspar) [reply tools/ci_build/github/apple/assemble_apple_packaging_artifacts.sh:29]: xcodebuild seem to be preserving link, but it seems to happen when assembling the pods

**Pull request merged**